lib/txMsgHelpers.ts

```diff
@@ -1,153 +1,60 @@
-  "value" in msg &&
-  "fromAddress" in msg.value &&
-  "toAddress" in msg.value &&
-  "amount" in msg.value &&
-  "endTime" in msg.value &&
-  "delayed" in msg.value &&
-  !!msg.value.fromAddress &&
-  !!msg.value.toAddress &&
-  !!msg.value.amount.length &&
-  !!msg.value.endTime &&
-  typeof msg.value.delayed === "boolean";
-
-const isTxMsgTransfer = (msg: TxMsg | EncodeObject): msg is TxMsgTransfer =>
-  msg.typeUrl === "/ibc.applications.transfer.v1.MsgTransfer" &&
-  "value" in msg &&
-  "sourcePort" in msg.value &&
-  "sourceChannel" in msg.value &&
-  "sender" in msg.value &&
-  "receiver" in msg.value &&
-  !!msg.value.sourcePort &&
-  !!msg.value.sourceChannel &&
-  !!msg.value.sender &&
-  !!msg.value.receiver;
-
-const gasOfMsg = (msgType: MsgType): number => {
+const gasOfMsg = (msgType: MsgTypeUrl): number => {
   switch (msgType) {
-    case "send":
+    case MsgTypeUrls.Send:
       return 100_000;
-    case "delegate":
+    case MsgTypeUrls.SetWithdrawAddress:
       return 100_000;
-    case "undelegate":
+    case MsgTypeUrls.WithdrawDelegatorReward:
       return 100_000;
-    case "redelegate":
+    case MsgTypeUrls.BeginRedelegate:
       return 100_000;
-    case "claimRewards":
+    case MsgTypeUrls.Delegate:
       return 100_000;
-    case "setWithdrawAddress":
+    case MsgTypeUrls.Undelegate:
       return 100_000;
-    case "createVestingAccount":
+    case MsgTypeUrls.CreateVestingAccount:
       return 100_000;
-    case "msgTransfer":
+    case MsgTypeUrls.Transfer:
       return 100_000;
     default:
       throw new Error("Unknown msg type");
   }
 };
 
-const gasOfTx = (msgTypes: readonly MsgType[]): number => {
+export const gasOfTx = (msgTypes: readonly MsgTypeUrl[]): number => {
   const txFlatGas = 100_000;
   const totalTxGas = msgTypes.reduce((acc, msgType) => acc + gasOfMsg(msgType), txFlatGas);
   return totalTxGas;
 };
 
-const importMsgType = (msg: EncodeObject): EncodeObject => {
-  if (isTxMsgCreateVestingAccount(msg)) {
-    return { ...msg, value: { ...msg.value, endTime: Long.fromValue(msg.value.endTime) } };
-  }
+export const isKnownMsgTypeUrl = (typeUrl: string): typeUrl is MsgTypeUrl =>
+  Object.values(MsgTypeUrls).includes(typeUrl as MsgTypeUrl);
 
-  if (isTxMsgTransfer(msg)) {
-    return {
-      ...msg,
-      value: { ...msg.value, timeoutTimestamp: Long.fromValue(msg.value.timeoutTimestamp) },
-    };
+export const exportMsgToJson = (msg: EncodeObject): EncodeObject => {
+  if (isKnownMsgTypeUrl(msg.typeUrl)) {
+    return { ...msg, value: MsgCodecs[msg.typeUrl].toJSON(msg.value) };
   }
 
   return msg;
 };
 
-const importMsgTypes = (msgs: readonly EncodeObject[]): EncodeObject[] => msgs.map(importMsgType);
+const importMsgFromJson = (msg: EncodeObject): EncodeObject => {
+  if (isKnownMsgTypeUrl(msg.typeUrl)) {
+    const parsedValue = MsgCodecs[msg.typeUrl].fromJSON(msg.value);
+    return { ...msg, value: parsedValue };
+  }
```

Comment:
Let's throw an error here if the type is not known. Otherwise we might carry around invalid data.